### PR TITLE
feat(onebox): Collect more telemetry

### PR DESCRIPTION
**Pull request by fkling**
Part of https://linear.app/sourcegraph/issue/SRCH-1455/one-box-telemetry

Adds or updates the following events:

- onebox.intentCorrection:clicked
- oneBox.moreResults:clicked
- onebox.resultContext:selectAll
- onebox.resultContext:deselectAll
- onebox.resultContext:individualSelected
- onebox.resultContext:individualSelected
- onebox.filterModal:opened|closed
- onebox.filter:clicked

TODO:

- onebox.query:executed
- onebox.manualSearch:toggled
- onebox.searchResult:clicked

## Test plan

n/a

### CODE REVIEW

File: vscode/webviews/chat/cells/messageCell/assistant/SearchFilters.tsx [hunk 1]
```diff
@@ -40,9 +43,16 @@ export const SearchFilters = ({
 
     const onFilterSelect = useCallback(
         (filter: NLSSearchDynamicFilter) => {
+            telemetryRecorder.recordEvent('onebox.filter', 'clicked', {
+                metadata: {
+                    filterType: getTelemetryFilterType(filter),
+                },
+                privateMetadata: { value: filter.value },
+                billingMetadata: { product: 'cody', category: 'billable' },
```
Comment (fkling) [inline vscode/webviews/chat/cells/messageCell/assistant/SearchFilters.tsx:51]: The issue says that all events are "billing DAU events", but I'm not sure if that means that this metadata has to be included.

Comment (thenamankumar) [reply vscode/webviews/chat/cells/messageCell/assistant/SearchFilters.tsx:51]: No idea. cc: @alexromano @akalia25

**Comment by fkling** [comment]
I'd say we get in these events at least... not quite sure what to do about the billing stuff though.

**Review by vovakulikov — APPROVED** [review]

**Review by thenamankumar — APPROVED** [review]

File: vscode/webviews/chat/cells/messageCell/assistant/SearchFilters.tsx (hunk 1 above)
Comment (akalia25) [reply vscode/webviews/chat/cells/messageCell/assistant/SearchFilters.tsx]: Hey, yes, for billing DAU events, we would need to add `billingMetadata` to the event. The category for these events would 'billable' and the product would be `search`,  `cody`,  `insights`, etc. 

The category would be ' core ' in the case of a "core" event. This document has more [information](https://www.notion.so/sourcegraph/V2-telemetry-product-and-category-documentation-b4f932118ea5421587e6e0a0d90d78e1?pvs=4). I'd be curious, do you have any suggestions on how could we improve the linear ticket to be more clear? Perhaps linking it to the doc shared above?

**Pull request merged**